pkg/controller/applications/comp.go

```diff
@@ -29,3 +31,80 @@ func IsApplicationUpToDate(cr *v1alpha1.ApplicationParameters, remote *argocdv1a
 
 	return cmp.Equal(*cluster, remote.Spec, opts...) && maps.Equal(cr.Annotations, remote.Annotations) && slices.Equal(cr.Finalizers, remote.Finalizers)
 }
+
+// GetApplicationCondition returns the condition of the application based on its status
+// Ready State Matrix:
+// ┌──────────────────┬───────────────┬────────────┬──────────────┬───────────┐
+// │ Health Status    │ Sync Status   │ Operation  │ Ready State  │ Notes     │
+// ├──────────────────┼───────────────┼────────────┼──────────────┼───────────┤
+// │ ""               │ ""            │     *      │ Unavailable  │ No Status │
+// │ "Missing"        │     *         │     *      │ Available    │ Not Sync'd│
+// │ "Healthy"        │     *         │ Succeeded  │ Available    │ Deployed  │
+// │ "Healthy"        │     *         │ Failed     │ Unavailable  │ Op Failed │
+// │ "Degraded"       │     *         │     *      │ Unavailable  │ Unhealthy │
+// │ "Progressing"    │     *         │     *      │ Unavailable  │ Not Ready │
+// │ Unknown          │     *         │     *      │ Unavailable  │ Bad State │
+// └──────────────────┴───────────────┴────────────┴──────────────┴───────────┘
+//
+// Notes:
+// - Empty status ("") indicates status not yet propagated
+// - Sync status is tracked but does not affect Ready state
+func GetApplicationCondition(status *v1alpha1.ArgoApplicationStatus) xpv1.Condition {
```

Comment:
This feels kind of overcomplex imho. And we should only propagate the plain external status at best without too much interpretation.

Since we are already writing the status information into `status.atProvider` it should be enough to just set the condition to `Unvailable`, `Creating` or `Available` depending on the state without generating or joining error messages.

If the current status does not expose enough information we should add another property to `status.atProvider`.

Reply:
Its less about the matrix and more about the custom condition messages. We should keep the controller logic simple. I think part of that additional status information is already being done in https://github.com/crossplane-contrib/provider-argocd/pull/247

Reply:
> Its less about the matrix and more about the custom condition messages. We should keep the controller logic simple. I think part of that additional status information is already being done in #247

@MisterMX please see updated commit with all custom messaging removed.